include/aos/common/spaceallocator/spaceallocator.hpp

```diff
@@ -74,6 +74,15 @@ class SpaceAllocatorItf {
      */
     virtual RetWithError<UniquePtr<SpaceItf>> AllocateSpace(uint64_t size) = 0;
 
+    /**
+     * Resizes space.
+     *
+     * @param size size to resize.
+     * @param space[out] resized space.
+     * @return Error.
+     */
+    virtual Error Resize(uint64_t size, UniquePtr<SpaceItf>& space) = 0;
```

Comment:
Probably it should be method of SpaceItf? Otherwise, it should be renamed to 'Reallocate'.